### PR TITLE
examples: create an example for enabling and configuring retry #3020

### PR DESCRIPTION
**Pull request by yhyddr**
https://github.com/grpc/grpc-go/issues/3020

### CODE REVIEW

**Comment by yhyddr** [comment]
what should I do next , add a Documentation?

File: examples/features/retry/server/main.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+)
+
+var port = flag.Int("port", 50052, "port number")
+
+type failingServer struct {
+	mu sync.Mutex
+
+	reqCounter uint
+	reqModulo  uint
+	reqSleep   time.Duration
```
[the remark this answers is not in the document]
Comment (yhyddr) [reply examples/features/retry/server/main.go]: I delete reqSleep
and hard-code is means this?
image

[the remark this answers is not in the document]
Comment (dfawley) [reply examples/features/retry/server/main.go]: By hard-code I meant: delete the field from `failingServer` and use `codes.Unavailable` directly where you call `status.Errorf`.

[the remark this answers is not in the document]
Comment (yhyddr) [reply examples/features/retry/server/main.go]: OK,

**Pull request merged**